### ISSUE TITLE
Warehouses over-supply the workshops

### ISSUE THREAD

**Issue by npliashechnikov** [Keriew/augustus#416]
Hello,

During my playthrough, I noticed that when I order a warehouse to accept a resource and there are workshops using this resource, the warehouse will supply them indefinitely. For example, I have seen pottery workshops with 15-20 stored clay - but without the warehouse, a workshop will not accept more than 3 units from a clay pit.
Is it possible to make the supply behaviour uniform for the mining sources and warehouses or make it configurable?

Thank you! Got to love the version 3.0, the quality-of-life improvements are amazing.

**Comment by npliashechnikov** [comment]
In particular, this happens when there are no more space for the goods the workshops manufacture - they don't produce anything yet still the warehouse will send them resources.

**Comment by crudelios** [comment]
@Keriew @bvschaik I think this is a bug with the original C3 as well. Either that or I messed up when refactoring the distribution code.

**Comment by bvschaik** [comment]
This happens in the original as well. It happens when there are workshops that are not connected to the warehouse that contains the raw materials.

**Comment by crudelios** [comment]
Right, I think this should be kept for Julius but fixed in Augustus.

**Comment by crudelios** [comment]
@Keriew Should we fix this for v3.1? If I have the time I can have a look.

**Comment by Keriew** [comment]
If we can have it soon, I think we could add it for 3.1. I've tried solving it in the past, but performance was an issue. It should be easier now that we have separate lists for building types.
